sylt-std/src/sylt.rs

```diff
@@ -8,324 +9,226 @@ use sungod::Ra;
 use sylt_common::error::RuntimeError;
 use sylt_common::{Blob, RuntimeContext, Type, Value};
 
-#[sylt_macro::sylt_doc(dbg, "Writes the type and value of anything you enter", [One(Value(val))] Type::Void)]
-#[sylt_macro::sylt_link(dbg, "sylt_std::sylt")]
-pub fn dbg<'t>(ctx: RuntimeContext<'t>) -> Result<Value, RuntimeError> {
-    let values = ctx.machine.stack_from_base(ctx.stack_base);
-    if ctx.typecheck {
-        println!(
-            "{}: {:?}",
-            "TYPECHECK".purple(),
-            values.iter().map(Type::from).collect::<Vec<_>>(),
-        );
-    } else {
+sylt_macro::extern_function!(
+    "sylt_std::sylt",
+    atan2,
+    ? "Returns the angle of a direction - relative to +x, where the first argument is x",
+    -> "fn float, float -> float",
+    [Float(x), Float(y)] => {
+        Ok(Float(y.atan2(*x)))
+    },
+);
+
+sylt_macro::extern_function!(
+    "sylt_std::sylt",
+    dbg,
+    ? "Writes the type and value of anything you enter",
+    -> "fn #X -> void",
+    [Float(x), Float(y)] => {
         println!(
-            "{}: {:?}, {:?}",
-            "RUN".purple(),
+            "DBG: {:?}, {:?}",
             values.iter().map(Type::from).collect::<Vec<_>>(),
             values
```

Comment:
I'm a bit hesistant towards values being supplied by the macro with no indication here. (If I read this code correctly.)